src/snapred/meta/mantid/WorkspaceNameGenerator.py

```diff
@@ -81,26 +82,33 @@ class _WorkspaceNameGenerator:
 
     _templateRoot = "mantid.workspace.nameTemplate"
     _delimiter = Config[f"{_templateRoot}.delimiter"]
-    _runTemplate = Config[f"{_templateRoot}.run"]
-    _runTemplateKeys = ["runNumber", "auxiliary", "lite", "unit", "group"]
-    _diffCalInputTemplate = Config[f"{_templateRoot}.diffCal.input"]
-    _diffCalInputTemplateKeys = ["unit", "runNumber"]
-    _diffCalTableTemplate = Config[f"{_templateRoot}.diffCal.table"]
-    _diffCalTableTemplateKeys = ["runNumber", "version"]
-    _diffCalOutputTemplate = Config[f"{_templateRoot}.diffCal.output"]
-    _diffCalOutputTemplateKeys = ["unit", "runNumber", "version"]
-    _diffCalMaskTemplate = Config[f"{_templateRoot}.diffCal.mask"]
-    _diffCalMaskTemplateKeys = ["runNumber", "version"]
-    _diffCalMetricTemplate = Config[f"{_templateRoot}.diffCal.metric"]
-    _diffCalMetricTemplateKeys = ["metricName", "runNumber", "version"]
-    _diffCalTimedMetricTemplate = Config[f"{_templateRoot}.diffCal.timed_metric"]
-    _diffCalTimedMetricTemplateKeys = ["metricName", "runNumber", "timestamp"]
-    _rawVanadiumTemplate = Config[f"{_templateRoot}.normCal.rawVanadium"]
-    _rawVanadiumTemplateKeys = ["unit", "group", "runNumber"]
-    _focusedRawVanadiumTemplate = Config[f"{_templateRoot}.normCal.focusedRawVanadium"]
-    _focusedRawVanadiumTemplateKeys = ["unit", "group", "runNumber"]
-    _smoothedFocusedRawVanadiumTemplate = Config[f"{_templateRoot}.normCal.smoothedFocusedRawVanadium"]
-    _smoothedFocusedRawVanadiumTemplateKeys = ["unit", "group", "runNumber"]
+
+    def __init__(self):
+        self._setupTemplateVars(Config[f"{self._templateRoot}.template"])
+
+    def _setupTemplateVars(self, templateDict, namePrefix=""):
+        for key, value in templateDict.items():
+            name = key
+            if namePrefix:
+                name = f"{namePrefix}{self._capFirst(name)}"
+
+            if isinstance(value, str):
+                template = value
+                self._setTemplateVar(template, name)
+            else:
+                self._setupTemplateVars(value, name)
```

Comment:
Could you add a comment explaining that this recursive call is to handle the multiple layers of keys?  It took me some squinting to see that.